u19_pipeline/automatic_job/ephys_element_populate.py

```diff
@@ -1,25 +1,67 @@
 
-def populate(display_progress=True):
+    populate_settings = {'display_progress': display_progress, 
+                         'reserve_jobs': reserve_jobs, 
+                         'suppress_errors': suppress_errors}
 
-    populate_settings = {'display_progress': display_progress, 'reserve_jobs': False, 'suppress_errors': False}
-
-    print('\n---- Populate ephys.EphysRecording ----')
     ephys_element.EphysRecording.populate(**populate_settings)
 
-    print('\n---- Populate ephys.LFP ----')
+    for key in (recording_process.Processing * recording.Recording & \
+                            dict(recording_modality='electrophysiology', \
+                                 status_processing_id=7)).fetch('KEY'):
+
+        recording_id, fragment_number, recording_process_pre_path = \
+                                (recording_process.Processing & key).fetch1(
+                                                'recording_id', 
+                                                'fragment_number', 'recording_process_pre_path')
+    
+        precluster_param_list_id = (recording_process.Processing.EphysParams & \
+                                                key).fetch1('precluster_param_list_id')
+        
+        ephys_element.PreClusterTask.insert1(
+                                dict(recording_id=recording_id,
+                                     insertion_number=fragment_number,
+                                     precluster_param_list_id=precluster_param_list_id,
+                                     precluster_output_dir=recording_process_pre_path,
+                                     task_mode='load'))
+
+    ephys_element.PreCluster.populate(**populate_settings)
+
     ephys_element.LFP.populate(**populate_settings)
 
-    print('\n---- Populate ephys.Clustering ----')
+    for key in (recording_process.Processing * recording.Recording & \
+                            dict(recording_modality='electrophysiology', \
+                                 status_processing_id=7)).fetch('KEY'):
+
+        recording_id, fragment_number, recording_process_post_path = \
+                                (recording_process.Processing & key).fetch1(
+                                                'recording_id', 
+                                                'fragment_number', 'recording_process_post_path')
+
+        precluster_param_list_id, cluster_paramset_idx = \
+                            (recording_process.Processing.EphysParams & key).fetch1(
+                                                        'precluster_param_list_id', 'cluster_paramset_idx')
+
+        ephys_element.ClusteringTask.insert1(
+            dict(recording_id=recording_id, 
+                 insertion_number=fragment_number, 
+                 precluster_param_list_id=precluster_param_list_id,
+                 paramset_idx=cluster_paramset_idx,
+                 clustering_output_dir=recording_process_post_path+'/kilosort_output',
```

Comment:
> Is `recording_process_post_path` pulled as a string? If so, keep. I know elements sometimes retrieve `pathlib.Path()` objects which use `/` as a concatenation operator

Thanks.  In this case, it is stored within the table as a varchar datatype.

Reply:
> Hardcode `kilosort_output`? That will never change? Could be optional parameter of `run` if we think this could change

Good point.  I will confirm with Alvaro when we open a pull request to the `BrainCOGS` repository.